htdocs/multicurrency/class/multicurrency.class.php

```diff
@@ -558,9 +558,9 @@ public static function getAmountConversionFromInvoiceRate($fk_facture, $amount,
 
         if ($multicurrency_tx)
         {
-            if ($way == 'dolibarr') return $amount * $multicurrency_tx;
-            else return $amount / $multicurrency_tx;
-        } else return $amount;
+            if ($way == 'dolibarr') return (double) $amount * $multicurrency_tx;
```

Comment:
(double) generates a lot of trouble because it is a float type.
It would be better to use
price2num($amount * $multicurrency_tx, 'MT')

Reply:
this work after using price2num($amount * $multicurrency_tx, 'MU'). Thanks for the advise.